ibis/expr/types.py

```diff
@@ -46,7 +48,9 @@ def __repr__(self):
             return repr(result)
 
     def __hash__(self):
```

Comment:
can we use `@cached_property`?

Reply:
Not really. Python treats `__hash__` as a function so we cannot use @cached_property on this